code/API_definitions/qos-booking-and-assignment.yaml

```diff
@@ -0,0 +1,1756 @@
+          properties:
+            bookingId:
+              $ref: "#/components/schemas/BookingId"
+            totalDevicesInBooking:
+              $ref: "#/components/schemas/DeviceCount"     
+            remainingDevicesInBooking:
+              $ref: "#/components/schemas/DeviceCount"                             
+            qosProfile:
+              $ref: "#/components/schemas/QosProfileName"
+            startTime:
+              $ref: "#/components/schemas/Time"
+            duration:
+              $ref: "#/components/schemas/Duration"
+            serviceArea:
+              $ref: "#/components/schemas/Area"   
+            status:
+              $ref: "#/components/schemas/Status"
+            statusInfo:
+              $ref: "#/components/schemas/BookingStatusInfo"
+            statusSource:
+              $ref: "#/components/schemas/StatusSource"
+          required:
+            - bookingId
+            - totalDevicesInBooking
+            - remainingDevicesInBooking
+            - qosProfile
+            - startTime
+            - duration
+            - serviceArea
+            - status
+  
+    DeviceAssignmentInput:
+      description: | 
+        Array of devices that needs to be assigned to a booking. 
+        - The size of the array should be less than or equal to the devices that are remaining in the original booking. 
+        - At least one device identifier is required when a 2-legged access token is used.  
+        - In a 3-legged situation, only one device can be assigned at a time, and the device details are automatically generated from the access token
+        - In such a situation, 'Devices' array SHOULD NOT be provided.  
+      type: object
+      properties:
+        devices:
+          $ref: "#/components/schemas/Devices"
+        sink:
+          type: string
+          format: uri
+          description: The address to which events shall be delivered, using the HTTP protocol.
+          example: "https://endpoint.example.com/sink"
+        sinkCredential:
+          $ref: "#/components/schemas/SinkCredential" 
+  
+    DeviceAssignmentOutput :
+      description: |
+         This schema is used to return device information associated with the booking as an output.
+         - It will have the BookingId along with all the originally requested information.
+         - In some situations (ex: assignments using 3-legged access token) the device details should not be disclosed as per the privacy and other legal requirements. 
+         - For example:  
+           - If there are 10 devices assigned in a booking; but the GET operation returns only 7; It means 3 devices can't be disclosed
+           - The source of truth is in the following properties in the booking details: "totalDevicesInBooking" and "remainingDevicesInBooking".        
+           - In such a situation the operator can set  'statusInfo' as 'DEVICE_DISCLOSURE_ERROR' to clearly communicate the fact that some devices are not disclosed.

```

Comment:
I wouldn't call it DEVICE_DISCLOSURE_ERROR but DEVICE_DISCLOSURE_POLICY_APPLIED or similar. Instead of "operator can" let's choose one between "operator may/should/must", probably may or should at most.